.github/workflows/integrate.yaml

```
@@ -49,10 +54,16 @@ jobs:
         
   integration:
     name: Integration Tests
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-24.04
     steps:
     - name: Check out code
       uses: actions/checkout@v4
+
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v5.3.0
```

nitpick: newest version of the action:
```suggestion
      uses: actions/setup-python@v5.4.0
```